assets/model_monitoring/components/src/shared_utilities/store_url.py

```diff
@@ -234,7 +234,9 @@ def write_file(self, file_content: Union[str, bytes], relative_path: str = None,
         try:
             if isinstance(container_client, FileSystemClient):
                 with container_client.get_file_client(full_path) as file_client:
-                    return file_client.upload_data(file_content, overwrite)
+                    if not file_client.exists():
```

Comment:
can we refactor the code as:
```py
with _get_file_client(container_client, full_path) as file_client:
  return file_client.upload_data(...)
```
and _get_file_client() defined as:
```py
def _get_file_client(filesystem_client):
  file_client = filesystem_client.get_file_client(_client(full_path)
  if not file_client.exists():
    file_client = filesystem_client.create_file(full_path)
  return file_client
```
You use append_data for a new created file_client, I suppose we can also use upload_data?

Reply:
I tried to use this refactor but upload_data() is still an issue on writing to Gen2 storage. I have only got it to work with append_data() and flush_data() with the create_file() client method. But then for future operations after the file is created successsfully we can use upload_data() to modify/overwrite in the FileSystemClient. I'm not sure why this gap exists just that it does

Reply:
I'll stick to the current code that uses append and flush data. Do you have any other suggestions?

Reply:
ok, maybe add some comments to explain why we have to do in this way.

Reply:
Added comment